### PR TITLE
Write some coverage tests for dialog control

### LINKED ISSUE #513: Dialog coverage
Increase the dialog code coverage to 90%+.

### PR DESCRIPTION
**Pull request by sstoyanovIG**
Closes #513 

Increase coverage for dialog control

### CODE REVIEW

File: tests/unit/dialog/tests.html [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
     <script type="text/javascript" src="../../../src/js/modules/infragistics.util.js"></script>
     <script type="text/javascript" src="../../../src/js/modules/infragistics.util.jquery.js"></script>
+	<script type="text/javascript" src="../../../src/js/modules/infragistics.util.jquerydeferred.js"></script>
```
Comment (kdinev) [inline tests/unit/dialog/tests.html]: There's no longer such a file.

File: tests/unit/dialog/tests.html [hunk 2]
```diff
@@ -430,6 +447,214 @@
...
+                state: "maximized"
+            });
+
+            var dialog = $("#dialog6").data().igDialog;        
+
+            dialog.restore();
+            dialog.minimize();
+            dialog.restore();    
+
+            equal(dialog.options.state,"minimized","The dialog is restored");
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Umm, this actually looks like a minor bug 😕
Shouldn't the state be "opened"?

File: tests/unit/dialog/tests.html [hunk 3]
```diff
@@ -430,6 +447,214 @@
...
+
+            dialog.destroy();
+        }); 
+
+        test("Initizlize Container", function () {   
+            try 
+            {         
+                var objContainer = $("#newContainer")
+                $("#dialog5").igDialog("option", "container");
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Maybe this get call should be a check? Or just not needed?

File: tests/unit/dialog/tests.html [hunk 4]
```diff
@@ -430,6 +447,214 @@
...
+                $("#dialog5").igDialog("option", "container", "objContainer");
+                $("#dialog5").igDialog("option", "container", objContainer);
+
+                $.ig.TestUtil.keyInteraction(9, $("#dialog5").igDialog());
+            }
+            catch (e) 
+            { 
+                ok(false, "Container not set.") 
+            }
+            ok(true,"Container is set");
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: You can validate the container is set by getting it like so:
`ok($("#dialog5").igDialog("option", "container").is(objContainer), "Container is set");`
 Also is the try-catch needed really needed?

File: tests/unit/dialog/tests.html [hunk 5]
```diff
@@ -430,6 +447,214 @@
...
+            $("#dialog6").igDialog("option", "tabIndex", 1);
+            equal(dialog.options.tabIndex, 1,"The dialog's tab index is 1");
+
+            $("#dialog6").igDialog("option", "zIndex", 1);
+            equal(dialog.options.zIndex, 1,"The dialog's z index is 1");
+
+            $("#dialog6").igDialog("option", "trackFocus", true);
+            equal(dialog.options.trackFocus, true,"The dialog's trackFocus is true");
+
+            dialog.changeLocale();
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Any goal with the locale change?

File: tests/unit/dialog/tests.html [hunk 6]
```diff
@@ -430,6 +447,214 @@
...
+
+            simulateEvent(header, "mousedown", { which: 1, pageX: dim.right, pageY: vCenter });
+            simulateEvent(header, "mousemove", { which: 1, pageX: dim.right + 10, pageY: vCenter });
+            simulateEvent(header, "mouseup", { which: 1, pageX: dim.right, pageY: vCenter });
+
+            $("#dialog5").igDialog("option", "state", "maximized");
+
+            simulateEvent(header, "mousedown", { which: 1, pageX: dim.right, pageY: vCenter });
+            simulateEvent(header, "mousemove", { which: 1, pageX: dim.right + 10, pageY: vCenter });
+            simulateEvent(header, "mouseup", { which: 1, pageX: dim.right, pageY: vCenter });
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Maybe check whatever the events above were supposed to do happened?

File: tests/unit/dialog/tests.html [hunk 7]
```diff
@@ -430,6 +447,214 @@
...
+
+            $("#dialog5").igDialog("option", "state", "minimized");
+            simulateEvent(header, "dblclick", { which: 1, pageX: dim.right, pageY: vCenter });    
+            equal(dialog.options.state, "opened","The dialog is not opened after being minimized");  
+
+            simulateEvent(header, "touchstart", { which: 1, pageX: dim.right, pageY: vCenter, touches: 1});    
+            simulateEvent(header, "touchmove", { which: 1, pageX: dim.right + 10, pageY: vCenter });    
+            simulateEvent(header, "touchend", { which: 1, pageX: dim.right + 10, pageY: vCenter });    
+
+            $("#dialog5").igDialog("option", "showHeader", false);
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Also a check for this one?

File: tests/unit/dialog/tests.html [hunk 8]
```diff
@@ -430,6 +447,214 @@
...
+                dim = getElementDimensions($('#dialog5')),
+                vCenter = (dim.top + dim.bottom) / 2,
+                xCenter = (dim.left + dim.right) / 2;
+
+            simulateEvent(seResizer[0], "mousedown", { which: 1, pageX: dim.right, pageY: vCenter });
+            simulateEvent(seResizer[0], "mousemove", { which: 1, pageX: dim.right - 100, pageY: vCenter});
+            simulateEvent(seResizer[0], "mouseup", { which: 1, pageX: dim.right - 10, pageY: vCenter });
+
+            var resultDim = getElementDimensions($('#dialog5'));
+            equal(resultDim.right, dim.right,"The dialog is not opened after being minimized");
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Shouldn't it have moved instead to something like `dim.right - 100`?

File: tests/unit/dialog/tests.html [hunk 9]
```diff
@@ -430,6 +447,214 @@
...
+            simulateEvent(seResizer[0], "mousedown", { which: 1, pageX: dim.right, pageY: vCenter });
+            simulateEvent(seResizer[0], "mousemove", { which: 1, pageX: dim.right - 100, pageY: vCenter});
+            simulateEvent(seResizer[0], "mouseup", { which: 1, pageX: dim.right - 10, pageY: vCenter });
+
+            var resultDim = getElementDimensions($('#dialog5'));
+            equal(resultDim.right, dim.right,"The dialog is not opened after being minimized");  
+
+            simulateEvent(seResizer[0], "mousedown", { which: 1, pageX: dim.right, pageY: vCenter });
+            simulateEvent(seResizer[0], "mousemove", { which: 1, pageX: dim.right , pageY: vCenter + 1000});
+            simulateEvent(seResizer[0], "mouseup", { which: 1, pageX: dim.right - 10, pageY: vCenter });
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: A check after this one

File: tests/unit/dialog/tests.html [hunk 10]
```diff
@@ -430,6 +447,214 @@
...
+        test("InputName", function () {            
+            $("#dialog6").igDialog({     
+                inputName: "input1",
+                position: [100,100]
+            });
+
+            $("#dialog6").igDialog("option", "inputName", "input2");
+            $("#dialog6").igDialog("option", "state", "minimized");
+
+            ok($("#input1")[0].value.indexOf("s1:w300") !== -1,"The saved value is correct");
```
Comment (damyanpetev) [inline tests/unit/dialog/tests.html]: Maybe also check the state contains `p100,100` matching the initial options too :)

**Pull request merged**